kubeyard/commands/shell.py

```diff
@@ -47,7 +47,18 @@ def run_default(self):
     @cached_property
     def pod_name(self) -> str:
         if self.pod:
-            return self.pod
+            all_pods = sh.kubectl.get.pods('-o', 'jsonpath={.items[*].metadata.name}').split()
+            # Exact match
+            if self.pod in all_pods:
+                return self.pod
+            # Starting-with match
+            pods = [pod for pod in all_pods if pod.startswith(self.pod)]
+            if len(pods) == 1:
+                return pods[0]
+            elif len(pods) == 0:
+                raise CommandException(f"Not found pod equal or starting with '{self.pod}'")
+            else:
+                raise CommandException(f"Found more than one pod starting with '{self.pod}'")
```

Comment:
In order to resolve cases like:
`kubeyard shell -p my-app-696f4d7fdd-pgm7l`
`kubeyard shell -p my-app-additional-functionality-696f4d7fdd-pgm7l`
where providing  `kubeyard shell -p my-app` doesn't work

I propose following changes:
```suggestion
            pods.sort(key=len)
            if len(pods) == 0:
                raise CommandException(f"Not found pod equal or starting with '{self.pod}'")
            elif len(pods) > 1:
                logger.warning(f"Found more than one pod. Using '{pods[0]}'")
            return pods[0]
```

Reply:
That's a nice enhancement for me.

Shall I push that changes?

Reply:
Yes, please